browser/ai_chat/enabled_state_transition_service.cc

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/ai_chat/enabled_state_transition_service.h"
+
+#include "base/check.h"
+#include "brave/components/ai_chat/core/browser/utils.h"
+#include "brave/components/ai_chat/core/common/pref_names.h"
+#include "brave/components/constants/webui_url_constants.h"
+#include "chrome/browser/profiles/profile.h"
+#include "components/user_prefs/user_prefs.h"
+#include "content/public/browser/browser_context.h"
+#include "content/public/browser/web_ui_data_source.h"
+
+#if !BUILDFLAG(IS_ANDROID)
+#include <algorithm>
```

Comment:
most of these includes do not belong behind `!IS_ANDROID` guard. Guards in gn and cpp should always match.